app/models/miq_report/generator.rb

```diff
@@ -191,127 +191,141 @@ def generate_table(options = {})
+              elsif performance
+                generate_performance_results(options)
+              elsif interval == 'daily' && db_klass <= MetricRollup
+                generate_daily_metric_rollup_results(options)
+              elsif interval
+                generate_interval_metric_results(options)
+              else
+                generate_basic_results(options)
+              end
 
-    klass = db.kind_of?(Class) ? db : Object.const_get(db)
-    interval = db_options.present? && db_options[:interval]
-    custom_results_method = (db_options && db_options[:rpt_type]) ? "build_results_for_report_#{db_options[:rpt_type]}" : nil
-
-    includes = get_include_for_find
-
-    load_custom_attributes
-
-    time_profile.tz ||= tz if time_profile # Default time zone in profile to report time zone
-    ext_options = {:tz => tz, :time_profile => time_profile}
-    # TODO: these columns need to be converted to real SQL columns
-    # only_cols = cols
-    self.extras ||= {}
+    if db_options && db_options[:long_term_averages] && results.first.kind_of?(MetricRollup)
+      # Calculate long_term_averages and save in extras
+      extras[:long_term_averages] = Metric::LongTermAverages.get_averages_over_time_period(results.first.resource, db_options[:long_term_averages].merge(:ext_options => ext_options))
+    end
 
-    if custom_results_method
-      if klass.respond_to?(custom_results_method)
-        # Use custom method in DB class to get report results if defined
-        results, ext = klass.send(custom_results_method, db_options[:options].merge(:userid      => options[:userid],
-                                                                                    :ext_options => ext_options,
-                                                                                    :report_cols => cols))
-      elsif self.respond_to?(custom_results_method)
-        # Use custom method in MiqReport class to get report results if defined
-        results, ext = send(custom_results_method, options)
-      else
-        raise _("Unsupported report type '%{type}'") % {:type => db_options[:rpt_type]}
-      end
-      # TODO: results = results.select(only_cols)
-      self.extras.merge!(ext) if ext && ext.kind_of?(Hash)
-
-    elsif performance
-      # Original C&U charts breakdown by tags
-      if performance[:group_by_category] && performance[:interval_name]
-        results, self.extras[:interval]  = db_class.vms_by_category(performance)
-        build_table(results, db, options)
-      else
-        results, self.extras[:group_by_tag_cols], self.extras[:group_by_tags] = db_class.group_by_tags(
-          db_class.find_entries(ext_options).where(where_clause).where(options[:where_clause]),
-          :category     => performance[:group_by_category],
-          :cat_model    => options[:cat_model],
-          :include      => includes
-        )
-        build_correlate_tag_cols
-      end
+    build_apply_time_profile(results)
```

Comment:
would have been nice to get this out of here as well.

(reiterate previous comment)